src/compiler/factory.ts

```diff
@@ -3286,7 +3295,17 @@ namespace ts {
                     createVariableStatement(
                         /*modifiers*/ undefined,
                         createVariableDeclarationList(
-                            flattenParameterDestructuring(parameter, temp, visitor, convertObjectRest)
+                            flattenDestructuringBinding(
+                                context,
+                                parameter,
+                                temp,
+                                /*skipInitializer*/ convertObjectRest,
+                                /*recordTempVariablesInLine*/ true,
+                                convertObjectRest

```

Comment:
This is not an issue now that I've merged in the work from `isolateObjectSpread`.